oak_runtime/src/io.rs

```diff
@@ -22,13 +22,23 @@ use oak_abi::{ChannelReadStatus, OakStatus};
 use oak_io::{Decodable, Encodable, OakError};
 pub use oak_io::{Receiver, Sender};
 
+/// Wrapper for storing a decodable message and the size of the original encoded version.
+pub struct MessageWithSize<T> {
+    pub message: T,
+    pub size: usize,
+}
+
```

Comment:
Why is this necessary? And what happens if the value of the size field does not match the actual size? AFAICT this is only used for metrics, and I feel the extra complexity is not warranted. We can look into the message and see its size there, if needed?

Reply:
This was to be able to use a Receiver rather than manually wait on channels, receive and decode the message to simplify the logic in the GrpcResponseIterator. I assumed there might be a need in future other cases might als use the size of the messages for monitoring, so thought it would be useful to make it possible to use a Receiver in these cases.

I am not sure what you mean by looking into the message and see its size, but I assume you mean go back to manually waiting on the channels and reading the raw message. Is that correct?

Reply:
As discused separately, removed this and used the size of the rsp_msg field rather thatn the size of the entire GrpcResponse proto for monitoring.